### PR TITLE
[chore] Upgrade datadog-agent and mapping go versions and fix breaking changes

### PR DESCRIPTION
**Pull request by songy23**
**Description:** 
Roll forward https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/24552 with stable versions of dependencies.

> Upgrade dependency versions of github.com/DataDog/opentelemetry-mapping-go and github.com/DataDog/datadog-agent and fix breaking changes in protobuf (context: https://github.com/DataDog/datadog-agent/pull/10650, https://github.com/DataDog/opentelemetry-mapping-go/pull/126).

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: cmd/configschema/go.mod [hunk 1]
```diff
@@ -1111,6 +1112,8 @@ replace github.com/openshift/api v3.9.0+incompatible => github.com/openshift/api
 // It appears that the v0.2.0 tag was modified.  Replacing with v0.2.1
 replace github.com/outcaste-io/ristretto v0.2.0 => github.com/outcaste-io/ristretto v0.2.1
 
+replace github.com/DataDog/datadog-agent/pkg/trace v0.47.0-rc.3 => github.com/DataDog/datadog-agent/pkg/trace v0.47.0-rc.3.0.20230720220350-143f7b52181b
```
Comment (mx-psi) [inline cmd/configschema/go.mod:1115]: Why do we need this?

Comment (songy23) [reply cmd/configschema/go.mod:1115]: `make update-otel` will unfortunately try to replace `v0.47.0-rc.3.0.20230720220350-143f7b52181b` with `v0.47.0-rc.3`. Unfortunately `v0.47.0-rc.3` doesn't contain the breaking change thus the downgrade performed by `update-otel` will always break datadog exporter. This blocked v0.82 so I had to revert the original PR.

Comment (mx-psi) [reply cmd/configschema/go.mod]: Discussed offline, I think we both feel like tagging a pre-release is a good option here

**Review by mx-psi — APPROVED** [review]

**Comment by mx-psi** [comment]
@dmitryax do you want to review before merging?

**Comment by mx-psi** [comment]
Needs a rebase after merging the `internal/datadog` module PR

**Review by dmitryax — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**